src/workerd/util/sqlite-kv.h

```diff
@@ -272,21 +276,12 @@ void SqliteKv::put(ArrayOfKeyValuePair& pairs, WriteOptions options) {
   // general structure can be shared somehow?
   auto& stmts = ensureInitialized(options.allowUnconfirmed);
   stmts.stmtMultiPutSavepoint.run({.allowUnconfirmed = options.allowUnconfirmed});
+
+  // If any of the puts throw an exception, rollback the transaction and re-throw the exception from
+  // the put that failed.
+  KJ_ON_SCOPE_FAILURE(rollbackMultiPut(stmts, options));
```

Comment:
Putting this at this scope now covers the `stmts.stmtMultiPutRelease.run()` call below, whereas previously the try-catch did not cover that statement. I'm not sure if that matters or not, though.

I don't have any problem with using a try-catch with bare `throw` here, by the way, if you find that more readable. Just factoring out the rollback bit into a helper function is a big improvement.